### PR TITLE
Update all non-test framework NuGets to latest version

### PR DESCRIPTION
**Pull request by drwill-ms**
I saved the MS test SDK updates for another PR.

### CODE REVIEW

File: e2e/test/iothub/messaging/MessageReceiveE2ETests.cs [hunk 1]
```diff
@@ -247,7 +247,7 @@ private async Task DoNotReceiveMessagesSentBeforeSubscriptionAsync(TestDeviceTyp
             {
                 await testDeviceCallbackHandler.WaitForReceiveMessageCallbackAsync(cts.Token).ConfigureAwait(false);
             };
-            receiveMessageOverCallback.Should().Throw<OperationCanceledException>();
+            await receiveMessageOverCallback.Should().ThrowAsync<OperationCanceledException>();
```
Comment (drwill-ms) [inline e2e/test/iothub/messaging/MessageReceiveE2ETests.cs:250]: There are tighter type validations done by fluent assertions now, so all these updates are to resolve compile errors.

File: e2e/test/iothub/service/QueryClientE2ETests.cs [hunk 2]
```diff
@@ -213,9 +213,9 @@ public async Task JobQuery_QueryWorks()
             // tests in this suite are written to work even if the queried job isn't the one they created.
             // That's why these checks aren't more specific.
             queriedJob.JobId.Should().NotBeNull();
-            queriedJob.JobType.Should().NotBeNull();
+            queriedJob.JobType.Should().NotBe(JobType.Unknown);
```
Comment (drwill-ms) [inline e2e/test/iothub/service/QueryClientE2ETests.cs:216]: Enums can't be null by default, so we should simply compare against the default enum value.

File: iothub/device/src/Microsoft.Azure.Devices.Client.csproj [hunk 3]
```diff
@@ -50,13 +50,13 @@
...
-    <PackageReference Include="MQTTnet" Version="4.0.0.167" />
+    <PackageReference Include="MQTTnet" Version="4.1.2.350" />
     <PackageReference Include="Microsoft.Azure.Amqp" Version="2.5.12" />
     <PackageReference Include="Microsoft.CodeAnalysis.NetAnalyzers" Version="6.0.0">
       <PrivateAssets>all</PrivateAssets>
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
     </PackageReference>
     <PackageReference Include="Newtonsoft.Json" Version="13.0.1" />
-    <PackageReference Include="Microsoft.Win32.Registry" Version="4.5.0" />
+    <PackageReference Include="Microsoft.Win32.Registry" Version="5.0.0" />
```
Comment (drwill-ms) [inline iothub/device/src/Microsoft.Azure.Devices.Client.csproj:60]: I don't know why we have some of these references, but for now I'm just upgrading them.

This one might be related to our user-agent string.

File: e2e/test/iothub/messaging/MessageReceiveE2ETests.cs (hunk 1 above)
Comment (abhipsaMisra) [inline e2e/test/iothub/messaging/MessageReceiveE2ETests.cs:250]: 👍🏽

**Pull request merged**